### PR TITLE
Added NXOS provider for tacacs_server_group

### PR DESCRIPTION
**Pull request by jonnytpuppet**
Exactly the same as radius_server_group, which has been merged

### CODE REVIEW

**Comment by jonnytpuppet** [comment]
tests/beaker_tests/tacacs_server_group/tacacs_server_group_provider_defaults.rb passed in 51.36 seconds
      Test Suite: tests @ 2015-11-05 14:37:57 +0000

**Comment by mikewiebe** [comment]
@jonnytpuppet Can you provide the beaker results, update the demo manifest and README doc.

File: lib/puppet/provider/tacacs_server_group/nxapi.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'cisco_node_utils' if Puppet.features.cisco_node_utils?
+begin
+  require 'puppet_x/cisco/autogen'
```
Comment (hunsbea) [inline lib/puppet/provider/tacacs_server_group/nxapi.rb]: Appears the autogen library isn't being used and this require can be removed

**Comment by jonnytpuppet** [comment]
tests/beaker_tests/tacacs_server_group/tacacs_server_group_provider_defaults.rb passed in 53.22 seconds
      Test Suite: tests @ 2015-11-12 14:34:11 +0000

**Comment by jonnytpuppet** [comment]
@mikewiebe @glennmatthews Can this be merged ASAP please? Thanks

File: README.md [hunk 2]
```diff
@@ -239,6 +240,7 @@ The following resources include cisco types and providers along with cisco provi
 * [`radius_server`](#type-radius_server)
 * [`syslog_server`](#type-syslog_server)
 * [`syslog_setting`](#type-syslog_setting)
+* [`tacacs_server_group (netdev_stdlib)`](#type-cisco_tacacs_server_group)
```
Comment (mikewiebe) [inline README.md]: remove the net_dev_stdlib label here.. it's already under a netdev heading.

**Comment by mikewiebe** [comment]
@jonnytpuppet One additional minor comment but after that it can be merged

**Comment by jonnytpuppet** [comment]
@mikewiebe done

**Pull request merged**